sycl-fusion/passes/kernel-fusion/SYCLSpecConstMaterializer.cpp

```diff
@@ -0,0 +1,313 @@
+//==-------------------- SYCLSpecConstMaterializer.cpp ---------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "SYCLSpecConstMaterializer.h"
+#include "llvm/IR/Constant.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/SYCLLowerIR/TargetHelpers.h"
+#include <llvm/ADT/StringRef.h>
+
+#define DEBUG_TYPE "sycl-spec-const-materializer"
+
+using namespace llvm;
+
+cl::opt<size_t> UseTestConstValues(
+    "sycl-materializer-debug-value-size",
+    cl::desc("Size of the spec const blob, debug use only."));
+
+const bool SYCLSpecConstMaterializer::IsDebug =
+    getenv("SYCL_MATERIALIZER_DEBUG");
+
+// When run through the JIT pipeline we have no way of using this pass' debug
+// type, hence the introduction of the environment variable above and the macro
+// below.
+#define MATERIALIZER_DEBUG(X)                                                  \
+  do {                                                                         \
+    if (IsDebug)                                                               \
+      X;                                                                       \
+    else                                                                       \
+      LLVM_DEBUG(X);                                                           \
+  } while (false)
+
+constexpr llvm::StringLiteral SPEC_CONST_DATA_NODE_NAME{"SYCL_SpecConst_data"};
```

Comment:
```suggestion
constexpr llvm::StringLiteral SpecConstDataNodeName{"SYCL_SpecConst_data"};
```

Adjust to naming conventions